propagation-stackdriver/src/test/java/zipkin2/propagation/stackdriver/StackdriverTracePropagationTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2016-2018 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.propagation.stackdriver;
+
+import brave.propagation.Propagation;
+import brave.propagation.TraceContext;
+import brave.propagation.TraceContextOrSamplingFlags;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class StackdriverTracePropagationTest {
+  private static final String XCLOUD_VALUE = "c108dc108dc108dc108dc108dc108d00";
+  private static final String B3_HEADER = "b3";
+  private static final String B3_TRACE_ID = "b3b3b3b3b3b34da6a3ce929d0e0e4736";
+  private static final String B3_VALUE = String.format("%s-00f067aa0ba902b7-1", B3_TRACE_ID);
+
+  private Propagation<String> propagation =
+      StackdriverTracePropagation.FACTORY.create(Propagation.KeyFactory.STRING);
+
+  @Test
+  public void b3TakesPrecedenceOverXCloud() {
+    TraceContext.Extractor<String> extractor =
+        propagation.extractor(new FakeGetter(XCLOUD_VALUE, B3_VALUE));
```

Comment:
I like the idea in general, but I think the FakeGetter makes it less clear, what's going on, vs just using a normal map.

For example, you can initialize an empty mutable map as a field named headers (ps no need to mark fields in a test as private as the keyword is cluttery). You can also initialize a map extractor as a field (ex propagation.extractor(Map::get))

Then, in each test use the actual values you want by setting them in the map. It will look more 

```
headers.put(TRACE_ID_NAME, XCLOUD_VALUE);

TraceContextOrSamplingFlags extracted = extractor.extract(headers);
```

Imho, it reads easier... less distractions, such as "what is the fake getter doing" and "what is this unused object?" It also is much more like normal usage.